pkg/machine/define/errors.go

```diff
@@ -40,3 +40,12 @@ type ErrNewDiskSizeTooSmall struct {
 func (err *ErrNewDiskSizeTooSmall) Error() string {
 	return fmt.Sprintf("invalid disk size %d: new disk must be larger than %dGB", err.OldSize, err.NewSize)
 }
+
+type ErrIncompatibleMachineConfig struct {
+	Name string
+	Path string
+}
+
+func (err *ErrIncompatibleMachineConfig) Error() string {
+	return fmt.Sprintf("incompatible machine config %q (%s) for this version of Podman", err.Path, err.Name)
```

Comment:
registry.SetExitCode(os.ENOTSUP) ?

Reply:
nope, we we also use this to emit a soft error ... the use case is like `podman machine ls`; as it is walking the dir for JSON files, it tries to read one; if we detect that this is an old config (version 0), we emit a soft error but continue on ... assuming we list another machine, it seems reasonable that the error code is 0 (meaning the ls worked).

In all other cases, where the VM name is provided (we are not walking a dir looking for files), then we do hard error and yes, in that case we could set the return code. 

i think we have options to explore, for example, if the code above worked, we could make the soft error just a fmt.errorf instead of calling this actual error (the message could be the same).  That way, again if the above worked, it would set the code.

thoughts?